### PR TITLE
fix init counter

### PR DESCRIPTION
**Pull request by junshi15**
the init count for worker solver is fixed.

### CODE REVIEW

File: caffe-distri/src/main/cpp/CaffeNet.cpp [hunk 1]
```diff
@@ -531,7 +531,10 @@ int CaffeNet<Dtype>::getInitIter(int solver_index) {
         return root_solver_->iter();
     else {
         CHECK(syncs_[solver_index]);
-        return syncs_[solver_index]->solver()->iter();
+	if (solver_index == 0)
```
Comment (anfeng) [inline caffe-distri/src/main/cpp/CaffeNet.cpp]: We will need some logic to handle CPU mode. 
https://github.com/yahoo/CaffeOnSpark/blob/master/caffe-distri/src/main/cpp/CaffeNet.cpp#L208-L218

`if (solver_index == 0 || solver_mode_ == Caffe::CPU)`

**Comment by yahoocla** [comment]
CLA is valid!

File: caffe-distri/src/main/cpp/CaffeNet.cpp (hunk 1 above)
Comment (junshi15) [inline caffe-distri/src/main/cpp/CaffeNet.cpp]: No need to add condition for CPU mode since syncs_ size either zero or one in that mode.

**Comment by anfeng** [comment]
+1

**Pull request merged**